app/Libraries/Traits/FilterTrait.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Libraries\Traits;
+
+use Illuminate\Support\Facades\Schema;
+
+trait FilterTrait
```

Comment:
a nghĩ em nên viết trait cho việc `where` thôi.
trong mỗi model e tạo 1 mảng chứa các `tên trường` và `toán tử`.
rồi check nếu request đưa lên có trường nào thì add where cho trường đó.